pkg/demoinfocs/game_events.go

```diff
@@ -516,16 +516,15 @@ func (geh gameEventHandler) playerBlind(data map[string]*msg.CSVCMsg_GameEventKe
 }
 
 func (geh gameEventHandler) flashBangDetonate(data map[string]*msg.CSVCMsg_GameEventKeyT) {
-	if geh.parser.isSource2() && !geh.parser.disableMimicSource1GameEvents {
-		return
-	}
 
 	nadeEvent := geh.nadeEvent(data, common.EqFlash)
 
 	geh.gameState().lastFlash.player = nadeEvent.Thrower
-	geh.dispatch(events.FlashExplode{
-		GrenadeEvent: nadeEvent,
-	})
+	if geh.parser.isSource2() && !geh.parser.disableMimicSource1GameEvents {
```

Comment:
not related. It's to update `geh.gameState().lastFlash.player` even with CS2 demos, so we should retrieve the thrower in other flashbang events.